multimodal-search-pdf/README.md

```diff
@@ -0,0 +1,129 @@
+- Customized executors to better fit the needs
+- Detailed docstrings for YAML files to help you understand Jina App
+
+- [PDF search](#pdf-search)
+  * [Use toy data](#use-toy-data)
+  * [Install](#install)
+  * [Run](#run)
+  * [Start the Server](#start-the-server)
+  * [Query via REST API](#query-via-rest-api)
+  * [Documentation](#documentation)
+  * [Community](#community)
+  * [License](#license)
+    
+
+## Use toy data
+
+We have included several PDF blogs as toy data in [`toy_data`](toy_data). This data is ready to use with this example.
+You can also use other PDF files supported by [`pdfplumber`](https://github.com/jsvine/pdfplumber).
+
+## Install
+
+```bash
+pip install -r requirements.txt
+```
+
+## Run
+
+| Command | Description |
+| :--- | :--- |
+| ``python app.py -t index`` | To index files/data |
+| ``python app.py -t query`` | To run query on the index |
+| ``python app.py -t dryrun`` | Sanity check on the topology |
+
+## Start the Server
+
+``` bash
+python app.py -t query_restful
+```
+
+## Query via REST API
+
+When the REST gateway is enabled, Jina uses the [data URI scheme](https://en.wikipedia.org/wiki/Data_URI_scheme) to represent multimedia data. Simply organize your picture(s) into this scheme and send a POST request to `http://0.0.0.0:45670/api/search`, e.g.:
+
+``` bash
+curl --request POST -d '{"top_k": 10, "mode": "search",  "data": ["jina hello multimodal"]}' -H 'Content-Type: application/json' 'http://0.0.0.0:45670/api/search'
+```
+
+[JSON payload syntax and spec can be found in the docs](https://docs.jina.ai/chapters/restapi/#).
+
+This example shows you how to feed data into Jina via REST gateway. By default, Jina uses a gRPC gateway, which has much higher performance and rich features. If you are interested in that, go ahead and check out our [other examples](https://learn.jina.ai) and [read our documentation on Jina IO](https://docs.jina.ai/chapters/io/#).
+
+## Understand the Flows
+
+The following image shows the structure of index flow, the text and image will be extracted from PDF files as chunks. Then we will use 3 paths to process and store the data.
+- For the first path, the DocIndexer will store the Document ID and Document data on disk
+- For the second path, the pods will filter the image chunks, and do the processing. It will also store chunk ID and chunk data on disk
+- For the third path, the pods will filter the text chunks, and further segment text into smaller chunks. The ChunkMeta indexer is used to store data at chunks level and text indexer is used to store data for chunks of chunks.
+The joiner will wait until 3 paths are finished.
+
+<p align="center">
+  <img src=".github/.README_images/indexflow.png?raw=true" alt="Jina banner" width="90%">
+</p>
+
+The following image shows the structure of query flow. For query, we can send text, image and PDF files to one flow and get the results. We use 2 parallel paths to process the query data and get results.
```

Comment:
two not 2

Reply:
Thanks! It's easy to forget these rules if not working on docs recently. 😂